src/software/new_geom/util/distance.h

```diff
@@ -12,3 +17,60 @@
  */
 double distance(const Line &first, const Point &second);
 double distance(const Point &first, const Line &second);
+
+/**
+ * Finds the shortest distance between two Points
+ *
+ * @param first Point
+ * @param second Point
+ * @return the shortest distance between first and second
+ */
+double distance(const Point &first, const Point &second);
+
+/**
+ * Finds the shortest distance between two Segments
+ *
+ * @param first Segment
+ * @param second Segment
+ * @return the shortest distance between first and second
+ */
+double distance(const Segment &first, const Segment &second);
+
+/**
+ * Finds the shortest distance between a Point and a Segment
+ *
+ * @param first the Point
+ * @param second the Segment
+ * @return the shortest distance between first and second
+ */
+double distance(const Point &first, const Segment &second);
+double distance(const Segment &first, const Point &second);
+
+/**
+ * Finds the shortest distance between a Point and a Polygon
+ *
+ * @param first the Point
+ * @param second the Polygon
+ * @return the shortest distance between first and second
+ */
+double distance(const Point &first, const Polygon &second);
+double distance(const Polygon &first, const Point &second);
+
+/**
+ * Finds the squared shortest distance between a Point and a Segment
+ *
+ * @param first the Point
+ * @param second the Segment
+ * @return the squared shortest distance between first and second
+ */
+double distanceSquared(const Point &first, const Segment &second);
+double distanceSquared(const Segment &first, const Point &second);
+
+/**
+ * Finds the squared shortest distance between two Points
+ *
+ * @param first Point
+ * @param second Point
```

Comment:
nit; might as well not put anything for the param description if we aren't actually putting any information (ie. instead of `@param first Point`, just `@param first`) same goes for the ones above.

Reply:
Done